### PR TITLE
Added OAM components handling

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

File: consul/consul.go [hunk 1]
```diff
@@ -29,3 +33,51 @@ func New(config config.Handler, log logger.Handler, kubeConfig config.Handler) a
 		adapter.Adapter{Config: config, Log: log, KubeconfigHandler: kubeConfig},
 	}
 }
+
+// ProcessOAM will handles the grpc invocation for handling OAM objects
+func (h *Consul) ProcessOAM(ctx context.Context, oamReq adapter.OAMRequest) (string, error) {
+	var comps []v1alpha1.Component
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply consul/consul.go:39]: @utkarsh-pro

[the remark this answers is not in the document]
Comment (utkarsh-pro) [reply consul/consul.go:39]: @leecalcote we are using OAM definition structs as they were defined in v1alpha1, we (more like I) haven't kept up with the later changes in the spec. We can't really escape specifying this here in my opinion.

[the remark this answers is not in the document]
Comment (leecalcote) [reply consul/consul.go:39]: Thanks for looking at this. When we bump versions, it'll be a find and replace task as opposed to a single location - that's fine. Good to confirm.

File: templates/oam/workloads/consulmesh.meshery.layer5io.schema.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "$id": "http://meshery.layer5.io/definition/Workload/ConsulMesh",
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "title": "ConsulMesh",
```
Comment (leecalcote) [inline templates/oam/workloads/consulmesh.meshery.layer5io.schema.json:4]: Please make display friendly.

**Pull request merged**